features/lwipstack/lwip/src/core/ipv4/lwip_dhcp.c

```diff
@@ -659,7 +660,7 @@ dhcp_handle_ack(struct netif *netif)
   for (n = 0; (n < LWIP_DHCP_PROVIDE_DNS_SERVERS) && dhcp_option_given(dhcp, DHCP_OPTION_IDX_DNS_SERVER + n); n++) {
     ip_addr_t dns_addr;
     ip_addr_set_ip4_u32(&dns_addr, lwip_htonl(dhcp_get_option_value(dhcp, DHCP_OPTION_IDX_DNS_SERVER + n)));
-    dns_setserver(n, &dns_addr);
+    dns_setserver(n, &dns_addr, _interface_name);
```

Comment:
Provide here the "netif" to dns module which then needs to check based on "netif" content to whether interface is multihoming or default.